plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -1115,7 +1123,7 @@ def _try_get_gas_estimate(self, transaction: JSONLike) -> Optional[int]:
                 transaction=cast(
                     TxParams, AttributeDictTranslator.from_dict(transaction)
                 ),
-                block_identifier=LatestBlockParam,
+                block_identifier="latest",
```

Comment:
I got this error after I update the method to raise on try

```
    raise TypeError("Could not encode to JSON: {}".format(exc))
TypeError: Could not encode to JSON: dict had unencodable value at keys: {'params': because (list had unencodable value at index: [1: because (Object of type _LiteralGenericAlias is not JSON serializable)])}
```

Which was caused by `LatestBlockParam` being an uninitialised object, replacing it with the string value fixed the issue

Reply:
This will fix the error that you have seen. Although, take into consideration that the reason why estimation fails might be because you have a bad tx. It might potentially waste funds.

Reply:
I know about the bad tx params, but I need this for the CLI tools for on-chain interaction. The end users might provide bad tx params sometimes and we need to make sure we raise at the right time